### DISCUSSION TITLE
At runtime, is schema information loaded from `node_modules/.prisma/client/schema.prisma`?

### DISCUSSION THREAD

**Discussion by ymc9** [prisma/prisma#19078] [Q&A] [answered]
Hi guys,

I have a very special use case where I want the Prisma-generated typing to be slightly different from the schema used at runtime.

I found that if I modify `node_modules/.prisma/client/schema.prisma` (like adding extra fields), the runtime behavior (the validation of payload structure) reflects my change. Just wondering if this is reliable (I know it's definitely not recommended) and if the `node_modules/.prisma/client/schema.prisma` is the single source of truth for runtime schema.

Thanks!

**Comment by jharrell** [comment]
Hey @ymc9! Your intuition is correct, this is not exactly something we'd like to endorse but yes, your schema is pulled from that path. 

I'd love to chat more though, even if this is a very-special-and-possibly-unique use case, maybe we can come up with some solutions that avoid manually modifying `node_modules`.

**Reply by ymc9** [reply]
Thank you for confirming this @jharrell !

I'm working on a Prisma-based toolkit called [ZenStack](https://github.com/zenstackhq/zenstack). It makes a few schema-level and runtime extensions to Prisma for supporting access control and a bunch of other new features.

The toolkit uses an extended DSL (a superset of Prisma schema) and transpiles down to Prisma schema. During the transpilation process, it generates a few "auxiliary fields" that are for internal use only. My goal is to "hide" these fields from the types generated by Prisma (so that developers don't see them in typing), but need to make them work at runtime. That's why I'm thinking of "injecting" them into `node_modules/.prisma/client/schema.prisma` after `prisma generate` completes.

I believe the use case is quite unique and hope I explained it clearly 😄

**Reply by jharrell** [reply, marked as answer]
Oh, duh, I should have looked at who I was talking with. 😅 ZenStack is super cool!

I'm going to get some additional clarity from the team but the solution here seems to be to obey the [output path](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/generating-prisma-client#using-a-custom-output-path) as that `schema.prisma` will be what the client will be using as it's source of truth. In this case it makes sense to `generate`, `<insert magic here>`, and then wait for bundling/deployment/etc.

**Reply by ymc9** [reply]
I'm glad that you're aware of our project and appreciate the recognition. Good to know that I'm heading in the right direction. Thank you!
